### DISCUSSION TITLE
osgEarth external project osgEarth_INCLUDE_DIR not defined problem

### DISCUSSION THREAD

**Discussion by yunusyurtturk** [pelicanmapping/osgearth#2617] [Support Q&A] [answered]
I build & install osgEarth to:
`D:\....\osgearth\build\install`
There are bin, include, lib and share directories as expected.

Now I'm using this template to start my application which will be using osgEarth:
https://github.com/gwaldron/osgearth/tree/master/share/ExternalProject

I set my `PREFIX` path, my CMake code is like below:

```
find_package(osgEarth CONFIG REQUIRED)
add_executable(${PROJECT_NAME}
                main.cpp
)
target_link_libraries(${PROJECT_NAME} PRIVATE osgEarth)
```

command can find the `osgEarthConfig.cmake` file under `/install/share` directory. However, when configuring/building my project, CMake complains:

`Could NOT find osgEarth (missing: osgEarth_INCLUDE_DIR)`

I dig into details and found:

`osgEarth_INCLUDE_DIR` variable is set in `osgEarthConfig.cmake` file like that:

```
set(XPREFIX OSGEARTH)
...
find_path(osgEarth_INCLUDE_DIR
    NAMES osgEarth/rtree.h
    HINTS ${${XPREFIX}_INCLUDE_DIRS}
)
```
So, the script is looking for `OSGEARTH_INCLUDE_DIRS` as hinted and this is the variable that is missing.

Am I supposed to provide those variables somehow before using osgEarth in my external project? (INCLUDE,LIBRARY,etc.)
I'm asking because, with the help of find_package(), I think those variables should be automatically resolved somehow, however my process complains about them being not defined. The other option is I'm doing something wrong or missing.

**Comment by gwaldron** [comment, marked as answer]
We have updated it in the master branch if you are willing and able to try that.

You're right -- the config setup in the latest release is not quite complete. I you can't use the master branch, I'm afraid you will need to specify some of those directories manually in CMake until we can cut a new patch release.
